src/model/proto/foreflight.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package foreflight;
+
+option optimize_for = CODE_SIZE;
+
+message ForeFlightChecklistContainer {
+  string type = 1; // Always "checklist"
+  ForeFlightChecklistPayload payload = 2;
+}
+
+message ForeFlightChecklistPayload {
+  string objectId = 1;
+  string schemaVersion = 2; // Always "1.0"
+  ForeFlightChecklistMetadata metadata = 3;
+  repeated ForeFlightChecklistGroup groups = 4; // Luckily always present and has 3 elements
+}
+
+// All metadata fields are optional!
+message ForeFlightChecklistMetadata {
+  optional string name = 1;
+  optional string detail = 2;
+  optional string tailNumber = 3;
+}
+
+message ForeFlightChecklistGroup {
+  string objectId = 1;
+  string groupType = 2;
+  repeated ForeFlightChecklistSubgroup items = 3; // This key might be missing altogether
+}
+
+message ForeFlightChecklistSubgroup {
+  string objectId = 1;
+  string title = 2;
+  repeated ForeFlightChecklist items = 3; // This key might be missing altogether
+}
+
+message ForeFlightChecklist {
+  string objectId = 1;
+  string title = 2;
+  repeated ForeFlightChecklistItem items = 3; // This key might be missing altogether
+}
+
+// This would have been better modelled by inheritance, or better still using oneOf.
+// Unfortunately, inheritance is not supported and oneOf cannot be used with repeated fields.
+message ForeFlightChecklistItem {
+  string objectId = 1;
```

Comment:
Proto field naming is usually object_id, group_type, etc. (and the generated code and the JSON field names should still be the same as you're using now)

https://protobuf.dev/programming-guides/proto3/#json

Reply:
Thanks for the hint! I missed the part about automagic mapping :( Fixed in 6425adc.